File: java/keycloak-sdk-core/src/main/java/io/github/xzawed/keycloak/core/Secrets.java
package io.github.xzawed.keycloak.core;
public final class Secrets {
  private Secrets() {}
  public static String mask(String value) {
    if (value == null || value.length() < 8) return "***";
    return value.substring(0, 3) + "***";
  }
  public static String maskBearer(String header) {
    if (header == null) return "***";
    return header.regionMatches(true, 0, "Bearer ", 0, 7) ? "Bearer ***" : "***";
  }
}

File: python/src/keycloak_sdk/_internal/secrets.py
"""시크릿 마스킹."""

from __future__ import annotations


def mask(value: str | None) -> str:
    # 완전 불투명 마스크: 접두 노출 없이 값의 존재 여부만 표시한다(있으면 "***",
    # 없으면 ""). 접두 3글자 노출은 보안 이득이 없고 짧은 시크릿에서 비율 유출이
    # 크므로 제거했다(Java Secrets.mask와 동형).
    return "***" if value else ""
